### PR TITLE
Add redux configuration and use redux for event handling.

### PR DESCRIPTION
**Pull request by jellebouwman**

### CODE REVIEW

File: components/timesheet/TimesheetContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  requestTimeEntriesSucces,
+  postTimesheetEntry,
+  postTimesheetEntrySucces
+}
+  from '../../ducks/time-entries';
+
+import Timesheet from './Timesheet';
+
+const mapStateToProps = state => ({
+  timesheetEntries: state.timesheetEntries.items
```
Comment (RikFrieling) [inline components/timesheet/TimesheetContainer.jsx]: the convention is to export this as a selector from your reducer. So in time-entries: 
export const timeEntriesSelector = (state) => state.timeEntries.items;
and here:
timesheetEntries: timeEntriesSelector(state)

File: components/timesheet/TimesheetContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  onRequestTimeEntriesSucces: requestTimeEntriesSucces,
+  onPostTimesheetEntry: postTimesheetEntry,
+  onPostTimesheetEntrySucces: postTimesheetEntrySucces
+}, dispatch);
+
+const TimesheetContainer = props => (
+  <Timesheet {...props} />
+);
+
+TimesheetContainer.propTypes = {
```
Comment (RikFrieling) [inline components/timesheet/TimesheetContainer.jsx]: you can write this as: 
TimesheetContainer.propTypes = Timesheet.propTypes;

File: components/timesheet-entry/TimesheetEntry.jsx [hunk 3]
```diff
@@ -6,28 +6,28 @@ import './timesheet-entry.scss';
...
   toggleDeleting = () => {
     this.setState(({ isEntryDeleting }) => ({ isEntryDeleting: !isEntryDeleting }));
   }
 
-  handleDelete = (event) => {
+  handleDelete = () => {
+    const { onDelete } = this.props;
+    const { id } = this.props;
+
+    onDelete(id);
```
Comment (boris-arkenaar) [inline components/timesheet-entry/TimesheetEntry.jsx]: You could really just write these three lines as:
```
this.props.onDelete(this.props.id);
```

File: components/timesheet/Timesheet.jsx [hunk 4]
```diff
@@ -1,34 +1,58 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 import TimesheetDate from '../timesheet-date/TimesheetDate';
 import TimesheetEntry from '../timesheet-entry/TimesheetEntry';
 import TimesheetInput from '../timesheet-input/TimesheetInput';
 
-import { fetchTimesheetEntries, postTimesheetEntry } from '../../services/fetch-timesheet-entries/fetch-timesheet-entries';
+import { fetchTimesheetEntries, postTimesheetEntry, deleteTimesheetEntry } from '../../services/fetch-timesheet-entries/fetch-timesheet-entries';
```
Comment (boris-arkenaar) [inline components/timesheet/Timesheet.jsx]: This line is quite long...

File: components/timesheet/TimesheetContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  timesheetEntries: timesheetEntriesSelector(state)
+});
+
+const mapDispatchToProps = dispatch => bindActionCreators({
+  onDeleteTimesheetEntry: deleteTimesheetEntry,
+  onDeleteTimesheetEntrySucces: deleteTimesheetEntrySucces,
+  onRequestTimeEntries: requestTimeEntries,
+  onRequestTimeEntriesSucces: requestTimeEntriesSucces,
+  onPostTimesheetEntry: postTimesheetEntry,
+  onPostTimesheetEntrySucces: postTimesheetEntrySucces
```
Comment (boris-arkenaar) [inline components/timesheet/TimesheetContainer.jsx]: I think, naming it like this, is quite right 👍

File: ducks/time-entries.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  error: '',
+  isLoading: false
+};
+
+export const timesheetEntriesSelector = state => state.timesheetEntries.items;
+
+export function timeEntriesReducer(state = initialState, action) {
+  switch (action.type) {
+    case DELETE_TIMESHEET_ENTRY:
+      return { ...state };
```
Comment (boris-arkenaar) [inline ducks/time-entries.js]: No loading state when deleting or posting?

File: pages/index.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import './index.scss';
 
 import Header from '../components/header/Header';
-import Timesheet from '../components/timesheet/Timesheet';
+import Timesheet from '../components/timesheet';
```
Comment (boris-arkenaar) [inline pages/index.js]: You are actually importing thte `TimesheetContainer` here.

**Pull request merged**